openpype/hosts/maya/plugins/publish/collect_render.py

```diff
@@ -69,388 +68,248 @@ class CollectMayaRender(pyblish.api.ContextPlugin):
-                    "strict_error_checking", True
-                )
-            }
-
-            # Collect Deadline url if Deadline module is enabled
-            deadline_settings = (
-                context.data["system_settings"]["modules"]["deadline"]
-            )
-            if deadline_settings["enabled"]:
-                data["deadlineUrl"] = render_instance.data.get("deadlineUrl")
-
-            if self.sync_workfile_version:
-                data["version"] = context.data["version"]
-
-                for instance in context:
-                    if instance.data['family'] == "workfile":
-                        instance.data["version"] = context.data["version"]
-
-            # handle standalone renderers
-            if render_instance.data.get("vrayScene") is True:
-                data["families"].append("vrayscene_render")
-
-            if render_instance.data.get("assScene") is True:
-                data["families"].append("assscene_render")
-
-            # Include (optional) global settings
-            # Get global overrides and translate to Deadline values
-            overrides = self.parse_options(str(render_globals))
-            data.update(**overrides)
-
-            # get string values for pools
-            primary_pool = overrides["renderGlobals"]["Pool"]
-            secondary_pool = overrides["renderGlobals"].get("SecondaryPool")
-            data["primaryPool"] = primary_pool
-            data["secondaryPool"] = secondary_pool
-
-            # Define nice label
-            label = "{0} ({1})".format(expected_layer_name, data["asset"])
-            label += "  [{0}-{1}]".format(
-                int(data["frameStartHandle"]), int(data["frameEndHandle"])
+            self.log.info(" -> attach render to: {}".format(s))
+
+        layer_name = layer.name()
+
+        # collect all frames we are expecting to be rendered
+        # return all expected files for all cameras and aovs in given
+        # frame range
+        layer_render_products = get_layer_render_products(layer.name())
+        render_products = layer_render_products.layer_data.products
+        assert render_products, "no render products generated"
+        exp_files = []
+        multipart = False
+        for product in render_products:
+            if product.multipart:
+                multipart = True
+            product_name = product.productName
+            if product.camera and layer_render_products.has_camera_token():
+                product_name = "{}{}".format(
+                    product.camera,
+                    "_" + product_name if product_name else "")
+            exp_files.append(
+                {
+                    product_name: layer_render_products.get_files(
+                        product)
+                })
+
+        has_cameras = any(product.camera for product in render_products)
+        assert has_cameras, "No render cameras found."
+
+        self.log.info("multipart: {}".format(
```

Comment:
Replace log string formatting to %-format instead? As you know https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/

Reply:
got you, makes sense, thanks for clarifying!